lib/common/compiler.h

```diff
@@ -128,7 +128,7 @@
 }
 
 /* vectorization */
-#if !defined(__clang__) && defined(__GNUC__)
+#if !defined(__clang__) && defined(__GNUC__) && __GNUC__ > 5
```

Comment:
1) What about `clang` ?
2) Can you point at some online doc where the version requirement is explained ?

Reply:
1. The `DONT_VECTORIZE` macro to disable the option is not needed on Clang.  This is described in the original commit https://github.com/facebook/zstd/commit/812e8f2a1 and covered correctly by the preexisting `#if !defined(__clang__)` portion of the condition.

2. https://gcc.gnu.org/gcc-4.4/changes.html:

> ### New Languages and Language specific improvements:
> **C family:**
> A new `optimize` attribute was added …

Here's a link to FreeBSD's [CI run (warning: large text)](https://ci.freebsd.org/job/FreeBSD-head-sparc64-build/14315/console) (we build with `-Werror`):

> /usr/src/sys/contrib/zstd/lib/common/zstd_internal.h:213: warning: 'optimize' attribute directive ignored

It seems Numpy [had a similar issue](https://github.com/numpy/numpy/issues/5450) on OpenBSD in the past (same 4.2.1 GCC).

Reply:
Thanks for sharing the details !

> https://gcc.gnu.org/gcc-4.4/changes.html:

So it's available since `gcc v4.4` then ?
Hence, should it rather be `__GNUC__ >= 5` ?

Reply:
Yes, I think since 4.4.  `>=` is fine with me.